jsonschema/src/keywords/all_of.rs

```diff
@@ -14,10 +14,12 @@ pub(crate) struct AllOfValidator {
 
 impl AllOfValidator {
     #[inline]
-    pub(crate) fn compile(items: &[Value], context: &CompilationContext) -> CompilationResult {
+    pub(crate) fn compile(items: &[Value], context: &mut CompilationContext) -> CompilationResult {
         let mut schemas = Vec::with_capacity(items.len());
         for item in items {
+            context.schema_path.push(item.to_string());
```

Comment:
Here we need indexes within `items` - e.g., apply `enumerate` to the `items` above and store indexes like strings. Similar applies to `anyOf` validator implementation below.